### PR TITLE
Implemented web config UI that updates config real time

### PR DESCRIPTION
**Pull request by Uzay-G**
random

This PR implements several features / ideas:

- adds a config editing page at `/config` to replace the need to directly edit the local file. This form is dynamically generated from the default config and the user config
- When you submit a change to this config page, it auto-updates the local config file and the current config being run on by Archivy. So for example, like in the gif, you can directly change the theme of the Archivy interface.

### CODE REVIEW

File: archivy/config.py [hunk 1]
```diff
@@ -72,14 +74,23 @@ def __init__(self):
...
         }
 
-    def override(self, user_conf: dict):
+    def override(self, user_conf: dict, nested_dict=None):
         for k, v in user_conf.items():
             # handle ES options, don't override entire dict if one key is passed
-            if k == "SEARCH_CONF":
-                for subkey, subval in v.items():
-                    self.SEARCH_CONF[subkey] = subval
+            if (nested_dict and not k in nested_dict) or (
```
Comment (edditler) [inline archivy/config.py:80]: Some comment might be nice here what the long if statement tests

File: archivy/templates/base.html [hunk 2]
```diff
@@ -67,6 +67,9 @@ <h3>{{ config.SITE_TITLE }}</h3>
             </div>
 
             {% if current_user.is_authenticated and not view_only %}
+		<a href="/config" alt="settings">
```
Comment (edditler) [inline archivy/templates/base.html:70]: As said, this might be cool to move someplace else.

File: archivy/templates/config.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		{% elif field.type in ["CSRFTokenField", "SubmitField"] %}
+			{{ field() }}
+		{% endif %}
+	{% endfor %}
+{%- endmacro -%}
+{% block content %}
+	<h1>Config</h1>
+	<p>Check out <a href="https://archivy.github.io/config" target="_blank" rel="noopener">the documentation</a> for more info on these options.</p>
+ 	<form action="/config" method="POST">
+		{{ render_form(conf, 1) }}
```
Comment (edditler) [inline archivy/templates/config.html:25]: I guess it would take some coding to include the nesting in the HTML

**Review by edditler — COMMENTED** [review]
To be honest, I wouldn't be able to build on this code quickly. A few more comments would be nice for sure, like a description why some function exists and how they are called. In any case I'm happy to merge this into production.

Thanks a lot!

**Comment by Uzay-G** [comment]
Alright! I implemented the feedback! Huge thanks @edditler :)

**Pull request merged**